app/frontend/components/pages/accounts/accountsDashboard.tsx

```diff
@@ -0,0 +1,296 @@
+  shouldShowSendTransactionModal,
+  shouldShowDelegationModal,
+  selectedAccountIndex,
+  totalWalletBalance,
+  totalRewardsBalance,
+  shouldShowConfirmTransactionDialog,
+  setLastAccountExplored,
+  isLastAccountExplored,
+  shouldShowTransactionErrorModal,
+  transactionSubmissionError,
+  closeTransactionErrorModal,
+}: Props) => {
+  const InfoAlert = () => (
+    <Fragment>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            <strong>Accounts</strong> offer a way to split your funds. You are able to delegate to
+            different pool from each account. Each account has different set of addresses and keys.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            Click explore/activate button to load data for related account. If you are using
+            hardware wallet, you will be requested to export public key.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account info">
+        <Alert alertType="warning sidebar">
+          <p>
+            This feature might not be supported on other wallets yet. If you decide to move your
+            funds to <strong>account</strong> other then <strong>account 0</strong>, you might not
+            see your funds in other wallets.
+          </p>
+        </Alert>
+      </div>
+    </Fragment>
+  )
+
+  const usedAccountsCount = accountsInfo.filter(({isUsed}) => isUsed).length
+
+  return (
+    <Fragment>
+      {shouldShowSendTransactionModal && <SendTransactionModal />}
+      {shouldShowDelegationModal && <DelegationModal />}
+      <div className="dashboard-column account">
+        <div className="card account-aggregated">
+          <div className="balance">
+            <div className="item">
+              <h2 className="card-title small-margin">Total balance</h2>
+              <div className="balance-amount">
+                {printAda(totalWalletBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+            <div className="item">
+              <h2 className="card-title small-margin">Total rewards balance</h2>
+              <div className="balance-amount">
+                {printAda(totalRewardsBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+          </div>
+          <div className="refresh-wrapper">
+            <button className="button secondary balance refresh" onClick={reloadWalletInfo}>
+              Refresh
+            </button>
+          </div>
+        </div>
+        <div className="mobile">
+          <InfoAlert />
+        </div>
+        <div className="accounts-wrapper">
+          <div className="dashboard-column account list">
+            <div>
+              {range(0, usedAccountsCount + 1).map((accountIndex) => (
```

Comment:
Looking at usedAccountsCount definition above:

```
const usedAccountsCount = accountsInfo.filter(({isUsed}) => isUsed).length
```

why not just do here:
```
{range(0, usedAccountsCount).map((accountIndex) => (
```
?

Because supposing this code does not crash, accountsInfo must contain even the unused account index as there is the line `account={accountsInfo[accountIndex]}` - doesn't it?

Reply:
If the concern are non-explored accounts when bulk key export is disabled, isn't it easier to deal in this while cycle with all accounts - used or not, but explored (i.e. present in accountInfos, if I understand correctly)

and then, if and only if the last explored account is used (i.e. the discovery was not done fully), add one more account tile with the index increased by one and all account info hidden and showing the "explore" button? And when you click the "explore" button, the account gets loaded into accounts info and the AccountsDashboard should be re-rended and if the newly explored account is unused, the additional tile just won't be rendered